File: RobWork/example/python/ex-invkin.py
from sdurw import *
import sys

WC_FILE = "/devices/serialdev/UR10e_2018/UR10e.xml"

if __name__ == '__main__':
    if len(sys.argv) != 2:
        print("Usage: python3 " + sys.argv[0] + " <path/to/RobWorkData>")
        sys.exit(1)

    wc = WorkCellLoaderFactory.load(sys.argv[1] + WC_FILE)
    if wc.isNull():
        raise Exception("WorkCell could not be loaded")
    device = wc.findSerialDevice("UR10e")
    if device.isNull():
        raise Exception("UR10e device could not be found.")

    defState = wc.getDefaultState()
    solver = ClosedFormIKSolverUR(device.cptr(), defState)

    Tdesired = Transform3Dd(Vector3Dd(0.2, -0.2, 0.5), EAAd(0, Pi, 0).toRotation3D())
    solutions = solver.solve(Tdesired, defState)

    print("Inverse Kinematics for " + device.getName() + ".")
    print(" Base frame: " + device.getBase().getName())
    print(" End/TCP frame: " + solver.getTCP().getName())
    print(" Target Transform: " + str(Tdesired))
    print("Found " + str(solutions.size()) + " solutions.")
    for solution in solutions:
        print(" " + str(solution))

File: RobWork/example/java/src/ExInvKin.java
import java.lang.String;
import org.robwork.LoaderRW;
import org.robwork.sdurw.*;
import static org.robwork.sdurw.sdurwConstants.Pi;


public class ExInvKin {
    public static final String WC_FILE =
            "/devices/serialdev/UR10e_2018/UR10e.xml";

    public static void main(String[] args) throws Exception {
        LoaderRW.load("sdurw");

        if (args.length != 1) {
            System.out.print("Usage: java " + ExInvKin.class.getSimpleName());
            System.out.println(" <path/to/RobWorkData>");
            System.exit(1);
        }

        WorkCellPtr wc = WorkCellLoaderFactory.load(args[0] + WC_FILE);
        if (wc.isNull())
            throw new Exception("WorkCell could not be loaded.");
        SerialDevicePtr device = wc.findSerialDevice("UR10e");
        if (device.isNull())
            throw new Exception("UR10e device could not be found.");

        State state = wc.getDefaultState();
        ClosedFormIKSolverUR solver = new ClosedFormIKSolverUR(
                device.asSerialDeviceCPtr(), state);

        final Transform3Dd Tdesired = new Transform3Dd(new Vector3Dd(0.2, -0.2, 0.5),
                (new EAAd(0, Pi, 0)).toRotation3D());
        final QVector solutions = solver.solve(Tdesired, state);

        System.out.println("Inverse Kinematics for " + device.getName() + ".");
        System.out.println(" Base frame: " + device.getBase().getName());
        System.out.println(" End/TCP frame: " + solver.getTCP().getName());
        System.out.println(" Target Transform: " + Tdesired.toString());
        System.out.println("Found " + solutions.size() + " solutions.");
        for(int i = 0; i < solutions.size(); i++) {
            System.out.println(" " + solutions.get(i).toString());
        }
    }
}
